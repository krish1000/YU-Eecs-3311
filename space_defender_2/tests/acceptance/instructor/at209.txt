--Test enemy projectile moves into friendly projectile, enemy projectile higher damage.
toggle_debug_mode
play(5,23,1,2,2,2,2)

setup_select(2)
setup_next(5)

move(A,1)
fire
pass
move(B,3)


