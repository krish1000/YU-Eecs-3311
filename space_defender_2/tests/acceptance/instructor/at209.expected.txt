  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,23,1,2,2,2,2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(2)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Spread
->setup_next(5)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:120/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[C,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
    A _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(A,1)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:112/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[A,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:10, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,23]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [A,1]
  Enemy Action:
  Natural Enemy Spawn:
    A Fighter(id:1) spawns at location [A,23].
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
    A S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  F
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->fire
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:106/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[A,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,20]
  Projectile:
    [-2,*]->damage:50, move:1, location:[A,2]
    [-3,*]->damage:50, move:1, location:[B,2]
    [-4,<]->damage:20, move:3, location:[A,19]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) fires at location [A,1].
      A friendly projectile(id:-1) spawns at location out of board.
      A friendly projectile(id:-2) spawns at location [A,2].
      A friendly projectile(id:-3) spawns at location [B,2].
  Enemy Action:
    A Fighter(id:1) gains 1 armour.
    A Fighter(id:1) moves: [A,23] -> [A,20]
      A enemy projectile(id:-4) spawns at location [A,19].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
    A S  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  <  F  _  _  _
    B _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:114/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[A,1]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,14]
  Projectile:
    [-2,*]->damage:50, move:1, location:[A,3]
    [-3,*]->damage:50, move:1, location:[C,3]
    [-5,<]->damage:100, move:10, location:[A,13]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [A,2] -> [A,3]
    A friendly projectile(id:-3) moves: [B,2] -> [C,3]
  Enemy Projectile Action:
    A enemy projectile(id:-4) moves: [A,19] -> [A,16]
  Starfighter Action:
    The Starfighter(id:0) passes at location [A,1], doubling regen rate.
  Enemy Action:
    A Fighter(id:1) moves: [A,20] -> [A,14]
      The Fighter collides with enemy projectile(id:-4) at location [A,16], healing 20 damage.
      A enemy projectile(id:-5) spawns at location [A,13].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
    A S  _  *  _  _  _  _  _  _  _  _  _  <  F  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
->move(B,3)
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:60/60, energy:106/120, Regen:1/4, Armour:2, Vision:12, Move:9, Move Cost:4, location:[B,3]
      Projectile Pattern:Spread, Projectile Damage:50, Projectile Cost:10 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,F]->health:150/150, Regen:5, Armour:11, Vision:10, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,11]
  Projectile:
    [-3,*]->damage:50, move:1, location:[D,4]
    [-5,<]->damage:50, move:10, location:[A,3]
    [-6,<]->damage:20, move:3, location:[A,10]
  Friendly Projectile Action:
    A friendly projectile(id:-2) moves: [A,3] -> [A,4]
    A friendly projectile(id:-3) moves: [C,3] -> [D,4]
  Enemy Projectile Action:
    A enemy projectile(id:-5) moves: [A,13] -> [A,3]
      The projectile collides with friendly projectile(id:-2) at location [A,4], negating damage.
  Starfighter Action:
    The Starfighter(id:0) moves: [A,1] -> [B,3]
  Enemy Action:
    A Fighter(id:1) moves: [A,14] -> [A,11]
      A enemy projectile(id:-6) spawns at location [A,10].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
    A _  _  <  _  _  _  _  _  _  <  F  _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  S  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  *  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _  _
